src/main/java/Duke.java

```diff
@@ -1,10 +1,146 @@
+                        printNew(taskList.get(taskList.size() - 1), taskList.size());
+                        break;
+                    case "deadline":
+                        String[] deadin = commands[1].split(" /by ", 2);
+                        if (deadin.length > 1) {
+                            taskList.add(new Task(deadin[0], deadin[1], 'D', false));
+                            printNew(taskList.get(taskList.size() - 1), taskList.size());
+                        }
+                        else {
+                            System.out.println("Formatting error");
+                        }
+                        break;
+                    case "event":
+                        String[] evin = commands[1].split(" /at ", 2);
+                        if (evin.length > 1) {
+                            taskList.add(new Task(evin[0], evin[1], 'E', false));
+                            printNew(taskList.get(taskList.size() - 1), taskList.size());
+                        }
+                        else {
+                            System.out.println("Formatting error");
+                        }
+                        break;
+                    case "help":
+                        printHelp(commands[1]);
+                        break;
+                    default:
+                        System.out.println("Input error");
+                }
+            }
+
+            // Command has a single parameter
+            else {
+                switch (curCommand) {
+                    case "list":
+                        System.out.println("Here are the tasks in your list:");
+                        for (int i = 0; i < taskList.size(); i++) {
+                            System.out.println((i + 1) + "." + taskToString(taskList.get(i)));
+                        }
+                        break;
+                    case "bye":
+                        System.out.println("Bye. Hope to see you again soon!");
+                        break;
+                    case "help":
+                        System.out.println("Type help (command) to show more.\nAvailable commands: done, todo, deadline, event, list, bye, help (displays this message)");
+                        break;
+                    default:
+                        System.out.println("Input error");
+                }
+            }
+            System.out.println(line);
+        }
+    }
+
+    // Checks if a string is a positive numeric value
+    public static int isPositiveNumeric(String str) {
+        try {
+            return Integer.parseInt(str);
+        } catch(NumberFormatException e){
+            return -1;
+        }
+    }
+
+    // Converts task to string
+    public static String taskToString(Task task) {
+        return "[" + task.getType() + "][" + (task.isDone() ? "X" : " ") + "] "
+                + task.getTask() + (task.getType() == 'T' ? "" : " (" + (task.getType() == 'D' ? "by: " : "at: ")+ task.getTrail() + ")");
```

Comment:
This parts seems a bit messy, perhaps you could add this as a method in your classes to make it easier to read